### ISSUE TITLE
php fatal error when Gallery returns nothing

### ISSUE THREAD

**Issue by Fen1kz** [splittingred/Gallery#240]
so if you place [[!Gallery]- in ur resource (parameters doesnt matter)
this is supposed to show all albums (and it shows ) but if you have request var (default is galAlbum) and set it to id which doesnt exist 
like /gallery?galAlbum=99 (assuming u doesnt have album with ID 99 ) page wont load and shows you fatal error:

Warning: Invalid argument supplied for foreach() in lalala/modx/core/cache/includes/elements/modsnippet/18.include.cache.php on line #103#

Fatal error: Cannot use string offset as an array in lalala/modx/core/cache/includes/elements/modsnippet/18.include.cache.php on line #2#

these lines is:
#103#(line 103 in original snippet) foreach ($data['items'] as $item) {
#162#(line 162 in original snippet) $placeholderPrefix.'id' => $data['album']['id'],

i've quick fixed by adding 

if (empty($data)) return '';

right after line 48: $data = $modx->call('galItem','getList',array(&$modx,$scriptProperties));

TLDR: if u call gallery with non-existing album - it breaks with fatal error on ur site
latest modx, latest gallery
there's a fix

**Comment by anhuth** [comment]
I had a similar problem with an album, which did exist, but was empty, i.e. did not contain any image. As a consequence, my jquery calls ( $(document).ready(function(){....} ) were destroyed. 
As temporary solution for this problem, the fix proposed above did not help at all. Instead I inserted:
   if (empty($data['items'])) return '';

**Comment by jacobkball** [comment]
This still hasn't been fixed, as I have a client who repeatedly comes up against this issue.

I'll try the fix, to see if that resolves things.

It would be good if Gallery could handle it, though.

**Comment by jacobkball** [comment]
I can confirm that this fix works, thank goodness.

Just to clarify - the extra string goes in the main Gallery snippet, for those who might wonder.

**Comment by jpdevries** [comment]
Gallery is now maintained at the modxcms fork here:
https://github.com/modxcms/Gallery/pulls

would anyone be interested in submitting a pull request there?
